pkg/tagger/collectors/ecs_main.go

```diff
@@ -43,6 +43,10 @@ type ECSCollector struct {
 
 // Detect tries to connect to the ECS agent
 func (c *ECSCollector) Detect(ctx context.Context, out chan<- []*TagInfo) (CollectionMode, error) {
+	if !config.IsFeaturePresent(config.Docker) {
+		return NoCollection, nil
+	}
```

Comment:
This collector relies on the ECS Agent not Docker, I think we need to have a dedicated `ECS EC2` feature?
I just think that assuming ECS EC2 == Docker is a bit risky, the container runtime and/or the socket path can change in the future

Reply:
I see:
```
// ECSCollector listen to the ECS agent to get ECS metadata.
// Relies on the DockerCollector to trigger deletions, it's not intended to run standalone
```

I did not check if it was actually the case, will check

Reply:
So, if I understood correctly, it appears that the ECSCollector is only a `Fetcher` that will trigger on miss and it relies on Docker `Streamer` to delete entities. So if the `Docker` collector is not running, we'll have a leak.

Reply:
thanks for confirming